src/sage/categories/finite_coxeter_groups.py

```diff
@@ -754,6 +754,152 @@ def permutahedron(self, point=None, base_ring=None):
+                        next_level.add(X)
+            from sage.combinat.posets.lattices import MeetSemilattice
+            return MeetSemilattice(data)
+
+        def coxeter_complex(self):
+            r"""
+            Return the Coxeter complex of ``self``.
+
+            Let `W` be a Coxeter group, and let `X` be the corresponding Tits
+            cone, which is constructed as the `W` orbit of the fundamental
+            chamber in the reflection representation. The *Coxeter complex*
+            of `W` is the simplicial complex `(X \setminus \{0\}) / \RR_{>0}`.
+            The face poset of this simplicial complex is given by the
+            :meth:`coxeter_poset()`. When `W` is a finite group, then the
+            Coxeter complex is homeomorphic to an `(n-1)`-dimensional sphere,
+            where `n` is the rank of `W`.
+
+            EXAMPLES::
+
+                sage: W = CoxeterGroup(['A', 3])
+                sage: C = W.coxeter_complex()
+                sage: C
+                Simplicial complex with 14 vertices and 24 facets
+                sage: C.homology()
+                {0: 0, 1: 0, 2: Z}
+
+                sage: W = WeylGroup(['B', 3])
+                sage: C = W.coxeter_complex()
+                sage: C
+                Simplicial complex with 26 vertices and 48 facets
+                sage: C.homology()
+                {0: 0, 1: 0, 2: Z}
+
+                sage: W = CoxeterGroup(['I', 7])
+                sage: C = W.coxeter_complex()
+                sage: C
+                Simplicial complex with 14 vertices and 14 facets
+                sage: C.homology()
+                {0: 0, 1: Z}
+
+                sage: W = CoxeterGroup(['H', 3])
+                sage: C = W.coxeter_complex()
+                sage: C
+                Simplicial complex with 62 vertices and 120 facets
+                sage: C.homology()
+                {0: 0, 1: 0, 2: Z}
+
+                sage: W = CoxeterGroup(['H', 3], implementation="permutation")  # optional - gap3
+                sage: C = W.coxeter_complex()                                   # optional - gap3
+                sage: C                                                         # optional - gap3
+                Simplicial complex with 62 vertices and 120 facets
+                sage: C.homology()                                              # optional - gap3
+                {0: 0, 1: 0, 2: Z}
+            """
+            I = self.index_set()
+            facets = {}
+            for g in self:
+                V = []
+                for i in I:
+                    gp = g
+                    D = gp.descents(side='right')
+                    if D and D[0] == i:
+                        D.pop(0)
+                    while D:
+                        gp = gp.apply_simple_reflection(D[0])
+                        D = gp.descents(side='right')
+                        if D and D[0] == i:
+                            D.pop(0)
+                    Ip = list(I)
```

Comment:
maybe store the tuples Ip[i] in a list before the main loop over self ?

Reply:
Good idea. Done.